### PR TITLE
novacustom_nv4x_adl: add heads section

### PR DESCRIPTION
**Pull request by macpijan**

### CODE REVIEW

**Comment by miczyg1** [comment]
Well, we probably don't need much more in documentation... So far looks good. Will try to adapt a similar approach to MSI

**Comment by mkopec** [comment]
We will probably need some documentation for changing the bootsplash, fortunately it'll just be a slightly different convert and cbfstool command than usual

**Comment by miczyg1** [comment]
> We will probably need some documentation for changing the bootsplash, fortunately it'll just be a slightly different convert and cbfstool command than usual

That will not be customizable logo, because replacing the bootsplash.jpg in CBFS will cause an removal of default logo... We may put the JPG/BMP into BOOTSPLASH region as previously and add support for looking into BOOTSPLASH region in coreboot bootsplash driver. Or just have a different CBFS filename for it, e.g. `bootsplash_custom.jpg/bmp`

**Comment by mkopec** [comment]
@BeataZdunczyk I overhauled the overview page, looks like this now:

image

**Comment by mkopec** [comment]
When following these build instructions on a freshly cloned repo, I run into https://github.com/linuxboot/heads/issues/1507

Doing `rm -rf build/x86/openssl-3.0.8/` as mentioned in https://github.com/linuxboot/heads/issues/1507#issuecomment-1767825980 works around this issue.

**Comment by mkopec** [comment]
Hmm, the openssl build error is still here. The buildsystem first removes openssl, then tries to find it again and errors out when it can't find it

**Comment by macpijan** [comment]
@mkopec What was the openssl error exactly? I remember having one as well back then. After changes in container, I removed the whole build and started build from scratch.

**Comment by mkopec** [comment]
@macpijan It was the exact same error as in https://github.com/linuxboot/heads/issues/1507#issuecomment-1753962680

**Comment by mkopec** [comment]
@macpijan I've added some of my debugging logs to the upstream issue: https://github.com/linuxboot/heads/issues/1507#issuecomment-1946736106

**Comment by mkopec** [comment]
Added release notes together with SBoM and a link to test results.

**Comment by mkopec** [comment]
Added NV4x w/ Heads tab to the USB-C HCL table and added a Heads content tab to the overview page. Updated the layout and wording of the layout page to make it more compact.

**Comment by mkopec** [comment]
I think this is ready for review now @miczyg1 @BeataZdunczyk

File: docs/unified/novacustom/overview.md [hunk 1]
```diff
@@ -1,57 +1,117 @@
 # Overview
 
-## Hardware
+Select your firmware flavor:
```
Comment (BeataZdunczyk) [inline docs/unified/novacustom/overview.md:3]:
```suggestion
Select your Dasharo firmware flavor:
```

File: docs/unified/novacustom/overview.md [hunk 2]
```diff
@@ -1,57 +1,117 @@
...
 
-=== "NS5x/7x 12th Gen"
-    The `NovaCustom NS5x/7x 12th Gen` section is related to the two physical
-    NovaCustom laptops based on the 12th generation Intel Core processor:
+=== "coreboot + UEFI"
```
Comment (BeataZdunczyk) [inline docs/unified/novacustom/overview.md:5]:
```suggestion
=== "Dasharo (coreboot + UEFI)"
```

File: docs/unified/novacustom/overview.md [hunk 3]
```diff
@@ -1,57 +1,117 @@
...
+    They both interact with each other tightly, so keeping their compatible versions
+    in sync is important. Information on compatibility should always be explained
+    on the release pages:
+
+    - [NV4x 12th Gen](/variants/novacustom_nv4x_adl/releases/)
+    - [NV4x 11th Gen](/variants/novacustom_nv4x_tgl/releases/)
+    - [NS5x/7x 12th Gen](/variants/novacustom_ns5x_adl/releases/)
+    - [NS5x/7x 11th Gen](/variants/novacustom_ns5x_tgl/releases/)
+
+=== "coreboot + Heads"
```
Comment (BeataZdunczyk) [inline docs/unified/novacustom/overview.md:79]:
```suggestion
=== "Dasharo (coreboot + Heads)"
```

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
+# Dasharo (coreboot + Heads) firmware variant
+
+For the NovaCustom NV4x 12th Gen, Heads-based variant of Dasharo firmware is
+offered as a Technology Preview Release.
+
+Please consider supporting the project financially by purchasing the
+[Dasharo Entry Subscription](https://novacustom.com/product/dasharo-entry-subscription/).
+With this subscription, you get access to the
+[Transition from Dasharo UEFI to Heads variant](#transition-from-dasharo-uefi-to-heads-variant)
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:9]:
```suggestion
[Transition from Dasharo UEFI to Heads variant](#switching-from-dasharo-uefi-to-heads)
```

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+`build/x86/nitropad-nv41/dasharo-nitropad-nv41-*.rom`.
+
+## Switching from Dasharo UEFI to Heads
+
+To change firmware branches from UEFI to Heads, because of how different the two
+firmware types are, it's required to disable some security measures before
+flashing. Follow the steps below to install Heads from an existing Dasharo UEFI
+firmware installation:
+
+* Enter UEFI Setup Menu and disable:
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:102]: I understand that Heads firmware  is designed for advanced users, but could you please share a quick guide on accessing the menu? I believe it's `F2` for NV4XPZ.

For naming consistency, users should head to `Dasharo System Features`  menu and then `Dasharo Security Options`  menu. Please verify, and use the exact names we have in our firmware.

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    - UEFI Secure Boot
+    - SMM BIOS Write Protection
+    - BIOS boot medium lock
+    - Intel Management Engine
+* [Boot into Dasharo Tools Suite](https://docs.dasharo.com/dasharo-tools-suite/documentation/#running)
+* Enter your DES subscription credentials
+* Select `Update Dasharo firmware` to check for updates
+* When asked to switch to Heads firmware, press `Y`
+* Proceed with the update steps as usual
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:111]: please link those steps

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+When the update is finished, your laptop with shut down automatically. Power it
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:113]:
```suggestion
When the update is finished, your laptop will shut down automatically. Power it
```

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+back on to boot into your new Heads installation!
+
+!!! warning "TOTP secrets warning"
+
+    On the first boot, you will be shown a warning about TOTP secrets. This is
+    normal and expected on the first boot. Run `OEM Factory Reset /
+    Re-Ownership` to finish deploying Heads.
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:120]: Please explain how to do that

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+!!! note "Note for Qubes OS users"
+
+    After installing Qubes while Heads is installed, you will need to select
+    `Reset TPM` in the Heads menu to finish the installation.
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:125]: > you will need to select  `Reset TPM`

where?

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+    Users upgrading to Heads while Qubes is already installed are not affected.
+
+## Switching from Dasharo Heads back to UEFI
+
+To revert back to UEFI, you will need to boot into DTS from a USB stick.
+
+TBD: Prepare docs for Heads-bootable DTS
```
Comment (BeataZdunczyk) [inline docs/variants/novacustom_nv4x_adl/heads.md:133]: TBD

**Review by BeataZdunczyk — CHANGES_REQUESTED** [review]

File: docs/unified/novacustom/hcl.md [hunk 11]
```diff
@@ -114,6 +114,28 @@ issue from which the report is sourced.
...
+        2. DP Alt mode: Synaptics VMM5310 DP MST hub, two upstream DP 1.4 lanes,
+           DSC 1.2 decompression, up to 2x 4K60 + 1x 4K30 depending on source.
+
+        ### Hubs
+
+        | Manufacturer | Model | Dasharo version | Notes | Source |
+        |---|---|---|---|---|
+        | Levin | 7-in-1 Type-C Hub Pro | v1.7.2 | No issues | Official support |
+        | Generic | 8-in-1 Type-C Hub | v1.7.2 | One video output usable at a time | Official support |
+
```
Comment (BeataZdunczyk) [inline docs/unified/novacustom/hcl.md:138]: Have we tested hubs to officially provide support for them?

File: docs/unified/novacustom/overview.md (hunk 1 above)
Comment (mkopec) [reply docs/unified/novacustom/overview.md]: Applied: https://github.com/Dasharo/docs/pull/754/commits/04010ad7df796e6220a0c02cacfe5e7b801734cf

File: docs/unified/novacustom/overview.md (hunk 2 above)
Comment (mkopec) [reply docs/unified/novacustom/overview.md]: Applied: https://github.com/Dasharo/docs/pull/754/commits/04010ad7df796e6220a0c02cacfe5e7b801734cf

File: docs/unified/novacustom/overview.md (hunk 3 above)
Comment (mkopec) [reply docs/unified/novacustom/overview.md]: Applied: https://github.com/Dasharo/docs/pull/754/commits/04010ad7df796e6220a0c02cacfe5e7b801734cf

File: docs/variants/novacustom_nv4x_adl/heads.md (hunk 4 above)
Comment (mkopec) [reply docs/variants/novacustom_nv4x_adl/heads.md:9]: Fixed: https://github.com/Dasharo/docs/pull/754/commits/b4dc330324793b94f91122d0a37b6c2cfc52bf2a

File: docs/variants/novacustom_nv4x_adl/heads.md (hunk 5 above)
Comment (mkopec) [reply docs/variants/novacustom_nv4x_adl/heads.md]: Added more explicit steps: https://github.com/Dasharo/docs/pull/754/commits/40abb8386a217b149422e72be135d21b82cdeed5

File: docs/variants/novacustom_nv4x_adl/heads.md (hunk 7 above)
Comment (mkopec) [reply docs/variants/novacustom_nv4x_adl/heads.md]: Applied: https://github.com/Dasharo/docs/pull/754/commits/f486a2bc14977bb9417900d712435b100a7b10dc

File: docs/variants/novacustom_nv4x_adl/heads.md (hunk 9 above)
Comment (mkopec) [reply docs/variants/novacustom_nv4x_adl/heads.md:138]: Explained here: https://github.com/Dasharo/docs/pull/754/commits/b4dc330324793b94f91122d0a37b6c2cfc52bf2a

File: docs/variants/novacustom_nv4x_adl/heads.md (hunk 6 above)
Comment (mkopec) [reply docs/variants/novacustom_nv4x_adl/heads.md]: Added a link: https://github.com/Dasharo/docs/pull/754/commits/3c463920e305094a8fef70b2cd6f2d2567c6611a

**Review by BeataZdunczyk — APPROVED** [review]

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+### Known issues
+
+* [Power button does not work in Qubes](https://github.com/Dasharo/dasharo-issues/issues/710)
+* [Heads shuts down instead of rebooting](https://github.com/Dasharo/dasharo-issues/issues/711)
+* [Existing Qubes installation is not bootable after transition back to EDK2](https://github.com/Dasharo/dasharo-issues/issues/713)
+
+### SBOM (Software Bill of Materials)
+
+* [Dasharo coreboot fork rev 3a9aa3a4 based on upstream version 4.21](https://github.com/Dasharo/coreboot/tree/3a9aa3a4)
+* [Dasharo Heads fork rev ccf49703 based on upstream rev 25066e5f](https://github.com/Dasharo/heads/tree/ccf49703)
```
Comment (miczyg1) [inline docs/variants/novacustom_nv4x_adl/heads.md:33]: This is not consitent with what I have in the  templates

Comment (miczyg1) [reply docs/variants/novacustom_nv4x_adl/heads.md]: Pushed an updated version: a549436d0ad196992a82db782a78b523ee78d379

**Review by BeataZdunczyk — APPROVED** [review]

File: docs/variants/novacustom_nv4x_adl/heads.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+subscription via Dasharo Entry Subscription newsletter.
+
+To verify binary integrity with hash and signature please follow the
+instructions in [Dasharo release signature verification](/guides/signature-verification)
+using [this key](https://github.com/3mdeb/3mdeb-secpack/blob/master/customer-keys/novacustom/dasharo-release-0.9.x-for-novacustom-signing-key.asc)
+
+### SBOM (Software Bill of Materials)
+
+- [Dasharo heads fork based on v0.2.0 revision ccf49703](https://github.com/Dasharo/heads/tree/ccf49703)
+- [Dasharo coreboot fork based on 4.21 revision 3a9aa3a4](https://github.com/Dasharo/edk2/tree/3a9aa3a4)
```
Comment (mkopec) [inline docs/variants/novacustom_nv4x_adl/heads.md:48]:
```suggestion
- [Dasharo coreboot fork based on 4.21 revision 3a9aa3a4](https://github.com/Dasharo/coreboot/tree/3a9aa3a4)
```

I guess an error in the scripts? @miczyg1

Comment (mkopec) [reply docs/variants/novacustom_nv4x_adl/heads.md]: fixed by fd6d63664d35

Comment (miczyg1) [reply docs/variants/novacustom_nv4x_adl/heads.md]: Yup, one sec

**Pull request merged**